src/metabase/models/revision.clj

```diff
@@ -98,12 +98,20 @@
   (cond
     (:is_creation revision)  [(deferred-tru "created this")]
     (:is_reversion revision) [(deferred-tru "reverted to an earlier version")]
+    (nil? prev-revision)     [(deferred-tru "modified this")]
     :else                    (diff-strings model (:object prev-revision) (:object revision))))
 
 (defn- revision-description-info
   [model prev-revision revision]
-  (let [changes (revision-changes model prev-revision revision)]
-    {:description          (build-sentence changes)
+  (let [changes     (revision-changes model prev-revision revision)
+        description (build-sentence changes)]
+    {:description          (if description
+                             description
+                             ;; HACK: before #30285 we record revision even when there is nothing changed,
+                             ;; so there are cases when revision can comeback as `nil`.
+                             ;; This is a safe guard for us to not display "Crowberto null" as
+                             ;; description on UI
+                             (deferred-tru "created a revision with no change."))
```

Comment:
I think the reasoning behind this would be better reflected in the code if you wrote the conditional in terms of `changes` rather than the `description`. Because otherwise to understand this code we'd need to understand all the reasons why `build-sentence` would return a nil description. Practically it returns a `nil` description iff `changes` is empty, so we might as well use that as the conditional directly
```suggestion
    {:description          (if (seq changes)
                             (build-sentence changes)
                             ;; HACK: before #30285 we record revision even when there is nothing changed,
                             ;; so there are cases when revision can comeback as `nil`.
                             ;; This is a safe guard for us to not display "Crowberto null" as
                             ;; description on UI
                             (deferred-tru "created a revision with no change."))
```

Reply:
updated in 254f33bedd